src/overlays/actors/ovl_En_Ig/z_en_ig.c

```diff
@@ -31,105 +70,951 @@ const ActorInit En_Ig_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF1FA8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF219C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2368.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2400.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2470.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF25E8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2890.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF293C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2A50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2AF8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2BD4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/EnIg_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/EnIg_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/EnIg_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2EDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF2EFC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/func_80BF302C.s")
+static ActorAnimationEntryS sAnimations[] = {
+    { &object_dai_Anim_0048B4, 1.0f, 0, -1, 0, 0 }, { &object_dai_Anim_0048B4, 1.0f, 0, -1, 0, -4 },
+    { &object_dai_Anim_005100, 1.0f, 0, -1, 0, 0 }, { &object_dai_Anim_005100, 1.0f, 0, -1, 0, -4 },
+    { &object_dai_Anim_0010F8, 1.0f, 0, -1, 2, 0 }, { &object_dai_Anim_001E44, 1.0f, 0, -1, 0, -4 },
+    { &object_dai_Anim_0014BC, 1.0f, 0, -1, 2, 0 }, { &object_dai_Anim_003CAC, 1.0f, 0, -1, 2, -4 },
+    { &object_dai_Anim_0040E0, 1.0f, 0, -1, 0, 0 }, { &object_dai_Anim_0040E0, 1.0f, 0, -1, 0, -4 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ig/EnIg_Draw.s")
+Actor* func_80BF1150(EnIg* this, GlobalContext* globalCtx, u8 actorCat, s16 actorId) {
+    Actor* foundActor = NULL;
+    Actor* temp_v0;
+
+    while (true) {
+        foundActor = SubS_FindActor(globalCtx, foundActor, actorCat, actorId);
+
+        if (foundActor == NULL) {
+            break;
+        }
+
+        if ((this != (EnIg*)foundActor) && (foundActor->update != NULL)) {
+            break;
+        }
+
+        temp_v0 = foundActor->next;
+        if (temp_v0 == NULL) {
+            foundActor = NULL;
+            break;
+        }
+        foundActor = temp_v0;
+    }
+
+    return foundActor;
+}
+
+EnDoor* func_80BF1200(GlobalContext* globalCtx, s32 arg1) {
+    s32 phi_a1;
+
+    switch (arg1) {
+        case 5:
+        case 6:
+        case 7:
+        case 9:
+            phi_a1 = 11;
+            break;
+
+        case 8:
+            phi_a1 = 15;
+            break;
+
+        default:
+            phi_a1 = -1;
+            break;
+    }
+    return (EnDoor*)SubS_FindDoor(globalCtx, phi_a1);
```

Comment:
```suggestion
    return SubS_FindDoor(globalCtx, phi_a1);
```
I was able to find a solution so that `SubS_FindDoor` actually returns an `EnDoor` now.